subprojects/gstreamer/libs/gst/base/gstaggregator.c

```diff
@@ -3206,24 +3206,26 @@ gst_aggregator_pad_chain_internal (GstAggregator * self,
         break;
       case GST_AGGREGATOR_START_TIME_SELECTION_FIRST:
         GST_OBJECT_LOCK (aggpad);
-        if (aggpad->priv->head_segment.format == GST_FORMAT_TIME) {
-          start_time = buf_pts;
-          if (start_time != -1) {
-            //GstClockTime buf_duration = GST_BUFFER_DURATION (buffer);
-            if (aggpad->priv->head_segment.rate < 0.0 && buf_duration != -1) {
-              start_time += buf_duration;
-            }
-            start_time = MAX (start_time, aggpad->priv->head_segment.start);
-            start_time =
-                gst_segment_to_running_time (&aggpad->priv->head_segment,
-                GST_FORMAT_TIME, start_time);
-          }
-        } else {
-          start_time = 0;
+        if (aggpad->priv->head_segment.format != GST_FORMAT_TIME) {
```

Comment:
what happens if we are currently playing in reverse or 2x ...
Do we know why we received buffer before segment ?

Approving not to block this PR, but not feeling very confident about the change.

Reply:
> Do we know why we received buffer before segment ?

We don't know why, that it would require investigation. What we know is that the fallback - what to do if the segment was not received, was not enough, so it was crashing in the videoaggregator class, that always expects some segment.